rust/src/tx_builder.rs

```diff
@@ -1077,13 +1142,21 @@ impl TransactionBuilder {
         }
     }
 
+    fn get_witness_set(&self) -> TransactionWitnessSet {
```

Comment:
We should be very explicit about what this includes and what doesn't. If you didn't look at the code you might not know it's only talking about the native scripts at all and might expect other witnesses. We have an issue #37 which would be nice to implement one day to help people sign txs after but as-is we don't have any of the info available to construct the other witnesses so unless we're very explicit about this function name (or at the very least documentation) I think it can be misleading.

Reply:
yeah this is a long-standing issue so I got the ball rolling by creating https://github.com/Emurgo/cardano-serialization-lib/pull/307

Reply:
We'll still want to document this even with #307 or will we restrict the witness builder API so it's a non-issue? (e.g. can't set native scripts after-the-fact so it must be done via the tx builder - or would we not want that?)